src/movies.js

```diff
@@ -1,25 +1,94 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies){
+  const directorsArray = movies.map(film => film.director);
+  return directorsArray;
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllFilteredDirectors(arr) {
+  const directorsWithDuplicates = movies.map(film => film.director);
+    const filtered = directorsWithDuplicates.filter(function(director, index){
+      return directorsWithDuplicates.indexOf(director) === index
+    })
+    return filtered
+  }
```

Comment:
Hi Anastasiia,

First of all let me thank you for writing such clean code with good naming that it is easy for me to read and understand. 🙏🏽. I will give you feedback to each iteration one at a time. If I skipped an iteration, then my only feedback would be: "good solution and clean code. 👌🏾"

**Here is my feedback to the first iteration:**
good solution but when we are working with automated tests, we don't change the tests to fit our solution. So please make all the work inside the provided function and do not add new modules to it.

Your approach is good. Here is how with small changes on your code you can make the tests pass without adding another function:

```
function getAllDirectors(movies){
//create a new array of directors
  const directorsArray = movies.map(function(movie) {
    return movie.director;
  })
// create a filtered array with no duplicates of the directors
  const filteredDirectorsArr = directorsArray.filter((value, index) => directorsArray.indexOf(value) === index);
  return filteredDirectorsArr;
}
```

Another way to remove duplicates:
`const filteredDirectorsArr = [...new Set(directorsArray)];`